src/wristpy/processing/metrics.py

```diff
@@ -53,3 +54,192 @@ def angle_relative_to_horizontal(
+) -> np.ndarray:
+    """Helper function to calculate the nonwear value array.
+
+    This function calculates the nonwear value array based on the GGIR 2023 methodology.
+    It computes the nonwear value for each axis, based on the acceleration data that
+    makes up one long epoch window. That nonwear value is then applied to all the
+    short windows that make up the long window. It iterates forward by one short_window
+    length and repeats the process. For the overlapping short windows, the maximum
+    nonwear value is kept and is assigned to the nonwear value array.
+
+    Args:
+        grouped_acceleration: The acceleration data grouped into short windows.
+        n_short_epoch_in_long_epoch: Number of short epochs that makeup one long epoch.
+        std_criteria: Threshold criteria for standard deviation.
+        range_criteria: Threshold criteria for range of acceleration.
+
+    Returns:
+        Non-wear value array.
+    """
+    nonwear_value_array = np.zeros(len(grouped_acceleration))
+
+    for window_n in range(len(grouped_acceleration) - n_short_epoch_in_long_epoch + 1):
+        acceleration_selected_long_window = grouped_acceleration[
+            window_n : window_n + n_short_epoch_in_long_epoch
+        ]
+
+        calculated_nonwear_value = acceleration_selected_long_window.select(
+            pl.col("X", "Y", "Z").map_batches(
+                lambda df: _compute_nonwear_value_per_axis(
+                    df, std_criteria, range_criteria
+                )
+            )
+        ).sum_horizontal()
+
+        max_window_value = np.maximum(
+            nonwear_value_array[window_n : window_n + n_short_epoch_in_long_epoch],
+            np.repeat(calculated_nonwear_value, n_short_epoch_in_long_epoch),
+        )
+        nonwear_value_array[window_n : window_n + n_short_epoch_in_long_epoch] = (
+            max_window_value
+        )
+
+    return nonwear_value_array
+
+
+def _compute_nonwear_value_per_axis(
+    axis_acceleration_data: pl.Series, std_criteria: float, range_criteria: float
+) -> int:
+    """Helper function to calculate the nonwear criteria per axis.
+
+    Args:
+        axis_acceleration_data: The long window acceleration data for one axis.
+            It is a pl.Series chunked into short windows where each row is a list of the
+            acceleration data of one axis (length of each list is the number of samples
+            that make up short_epoch_length in seconds).
+        std_criteria: Threshold criteria for standard deviation
+        range_criteria: Threshold criteria for range of acceleration
+
+    Returns:
+        Non-wear value for the axis.
+    """
+    axis_long_window_data = pl.concat(axis_acceleration_data, how="vertical")
+    axis_std = axis_long_window_data.std()
+    axis_range = axis_long_window_data.max() - axis_long_window_data.min()
+
+    criteria_boolean = (axis_std < std_criteria) & (axis_range < range_criteria)
+    return int(criteria_boolean)
```

Comment:
Why are we returning a boolean as an int?

Reply:
Python will actually do implicit conversions from bool to int when you try to do math with it e.g. `sum([True, True]) == 2`. So unless polars does some shenanigans here, you may be able to just remove the int cast without breaking anything else. 

If you do really need this as an int due to polars shenangians, I'd suggest converting it where you need it to make this clearer.

Reply:
Yeah, I just cast it because I do a polars.sum_horizontal() after and wasn't sure if it could handle booleans. Will check and convert if it works without casting.